packages/web3-providers-http/types/index.d.ts

```diff
@@ -0,0 +1,48 @@
+/*
+    This file is part of web3.js.
+
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+export class HttpProvider {
+    constructor(host: string, options?: HttpProviderOptions);
```

Comment:
The provider types are duplicated and not align with the 1.0 interface

Reply:
i will have to move them `web3-core-helpers` as base classes like the above which would mean they can be shared in `web3-core` and the provider classes.